extension/lib/doc.js

```diff
@@ -125,21 +134,80 @@ class Doc {
                     element: "readerease-sentence"
                 });
                 this.sentenceEls.push($("." + sentenceClassName));
+                // Set keywords and score for each sentence
+                this.setSentenceKeywordsAndScore(container, containerText, start, end, sentenceId);
             }
+            this.num_words_per_sentence = this.calcNumWordsPerSentence(this.sentences);
+            this.num_words_per_sentence_suffix_sum = suffix_sum(this.num_words_per_sentence);
             if (this.sentences.length > MAX_NUM_SENTENCE) {
                 alert("Sorry, we don't support big documents yet :(");
                 this.containers = [];
                 this.sentences = [];
                 this.containerIdToFirstSentenceId = [];
                 break;
             }
+            this.containerIdToFirstSentenceId.push(this.sentences.length);
```

Comment:
Why is this moved to the bottom, instead of letting it stay on the first few lines of the for loop?

** also not sure if we need containers anymore, but that's a separate refactoring cl

Reply:
Uncertain; moved back